app/helpers/application_helper.rb

```diff
@@ -1,2 +1,17 @@
+require 'net/http'
```

Comment:
Some thoughts:
1) Wow, pretty clever code there.
2) Never do this kind of fetching manually, especially in helpers.
3) If you want to play with gravatar try any gem, [this one](https://github.com/mdeering/gravatar_image_tag) for example 
4) This task is additional, so my suggestion is to remove that code cause I have no idea how it can be refactored well.